src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -3054,3 +3054,39 @@ class DropPolicyStatementSegment(BaseSegment):
         Ref("TableReferenceSegment"),
         OneOf("CASCADE", "RESTRICT", optional=True),
     )
+
+
+@postgres_dialect.segment(replace=True)
+class TruncateStatementSegment(BaseSegment):
+    """`TRUNCATE TABLE` statement.
+
+    https://www.postgresql.org/docs/14/sql-truncate.html
+    """
+
+    type = "truncate_table"
+    match_grammar = Sequence(
+        "TRUNCATE",
+        Ref.keyword("TABLE", optional=True),
+        Delimited(
+            OneOf(
+                Sequence(
+                    Ref.keyword("ONLY", optional=True),
+                    Ref("TableReferenceSegment"),
+                ),
+                Sequence(
+                    Ref("TableReferenceSegment"),
+                    Ref("StarSegment", optional=True),
+                ),
+            ),
+        ),
+        OneOf(
+            Sequence("RESTART", "IDENTITY"),
+            Sequence("CONTINUE", "IDENTITY"),
+            optional=True,
+        ),
```

Comment:
This could also be:

```suggestion
        Sequence(
            OneOf("RESTART", "CONTINUE"),
            "IDENTITY",
            optional=True,
        ),
```

But pretty much the same so a matter of personal preference.

Reply:
I tend to group the words as they're written in the docs Syntax diagram as IMO it's easier to compare the two but happy to make the change 👍 
![image](https://user-images.githubusercontent.com/80432516/147408286-aab68f7e-4105-48d2-9636-38f7b52f834a.png)